File: 2026POTD/JANUARY/15-01-2026/2943.swift
class Solution {
    func maximizeSquareHoleArea(_ n: Int, _ m: Int, _ hBars: [Int], _ vBars: [Int]) -> Int {
        func longest(_ a: [Int]) -> Int {
            let b = a.sorted()
            var best = 1, cur = 1
            for i in 1..<b.count {
                if b[i] == b[i - 1] + 1 {
                    cur += 1
                } else {
                    cur = 1
                }
                if cur > best {
                    best = cur
                }
            }
            return best + 1
        }

        let side = min(longest(hBars), longest(vBars))
        return side * side
    }
}

File: 2026POTD/JANUARY/15-01-2026/2943.php
class Solution {

    function maximizeSquareHoleArea($n, $m, $hBars, $vBars) {
        $longest = function(&$a) {
            sort($a);
            $best = 1;
            $cur = 1;
            for ($i = 1; $i < count($a); $i++) {
                if ($a[$i] == $a[$i - 1] + 1) {
                    $cur++;
                } else {
                    $cur = 1;
                }
                if ($cur > $best) {
                    $best = $cur;
                }
            }
            return $best + 1;
        };

        $side = min($longest($hBars), $longest($vBars));
        return $side * $side;
    }
}